tests/testthat/test-get-sys-info.R

```diff
@@ -0,0 +1,32 @@
+test_that("get_sys_info() returns date and system info", {
+  res <- get_sys_info()
+
+  expect_true(stringr::str_detect(res$date, as.character(Sys.Date())))
+  expect_equal(names(res$info$sys), c("sysname", "version", "release", "machine"))
+})
+
+test_that("get_sys_info() writes to json", {
+  tmp_file <- tempfile()
+  res <- get_sys_info(out_path = tmp_file)
+  withr::defer(unlink(tmp_file))
+
+  expect_true(fs::file_exists(tmp_file))
+
+  written_res <- jsonlite::fromJSON(tmp_file)
+  expect_true(stringr::str_detect(written_res$date, as.character(Sys.Date())))
```

Comment:
If you hit this at _just_ the right time, you'll get a spurious failure.  I dislike that, but I realize in practice it doesn't actually matter.  Still, I don't think we gain anything by testing against the actual `Date` value rather than just having a loose regexp check that the output looks date-like.

Reply:
you mean something simple like this? https://stackoverflow.com/a/15504877/3743213

No but really, I hear you. I'll change it to just check for `[0-9]+\\-[0-9]+\\-[0-9]+`. I don't think we need to validate that `Sys.time()` returns a valid date, just that it looks like `Sys.time()` was called and the results stored here.